compiler/rustc_resolve/src/imports.rs

```diff
@@ -1375,8 +1375,9 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                     target_bindings[ns].get(),
                 ) {
                     Ok(other_binding) => {
-                        is_redundant =
-                            binding.res() == other_binding.res() && !other_binding.is_ambiguity();
+                        is_redundant = binding.res() == other_binding.res()
```

Comment:
Or should we just suppress all redundant imports? 🤔 

That would be easier to throw under a new lint name if so.